internal/registry/clientsecretrequest/rest.go

```diff
@@ -70,32 +125,159 @@ func (*REST) Categories() []string {
 }
 
 func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	t := trace.FromContext(ctx).Nest("create", trace.Field{
-		Key:   "kind",
-		Value: "OIDCClientSecretRequest",
-	})
+	t := trace.FromContext(ctx).Nest("create",
+		trace.Field{
+			Key:   "kind",
+			Value: "OIDCClientSecretRequest",
+		},
+		trace.Field{
+			Key:   "metadata.name",
+			Value: name(obj),
+		},
+	)
 	defer t.Log()
 
-	_, err := validateRequest(obj, t)
+	req, err := r.validateRequest(ctx, obj, createValidation, options, t)
 	if err != nil {
 		return nil, err
 	}
+	t.Step("validateRequest")
+
+	oidcClient, err := r.clients.Get(ctx, req.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			traceValidationFailure(t, fmt.Sprintf("client %q does not exist", req.Name))
+			errs := field.ErrorList{field.NotFound(field.NewPath("metadata", "name"), req.Name)}
+			return nil, apierrors.NewInvalid(kindFromContext(ctx), req.Name, errs)
+		}
+		traceFailureWithError(t, "clients.Get", err)
+		return nil, apierrors.NewInternalError(fmt.Errorf("getting client %q failed", req.Name))
+	}
+	t.Step("clients.Get")
+
+	rv, hashes, err := r.secretStorage.Get(ctx, oidcClient.UID)
+	if err != nil {
+		traceFailureWithError(t, "secretStorage.Get", err)
+		return nil, apierrors.NewInternalError(fmt.Errorf("getting secret for client %q failed", req.Name))
+	}
+	t.Step("secretStorage.Get")
+
+	var secret string
+	if req.Spec.GenerateNewSecret {
+		secret, err = generateSecret(r.rand)
+		if err != nil {
+			traceFailureWithError(t, "generateSecret", err)
+			return nil, apierrors.NewInternalError(fmt.Errorf("client secret generation failed"))
+		}
+		t.Step("generateSecret")
+
+		hash, err := bcrypt.GenerateFromPassword([]byte(secret), cost)
+		if err != nil {
+			traceFailureWithError(t, "bcrypt.GenerateFromPassword", err)
+			return nil, apierrors.NewInternalError(fmt.Errorf("hash generation failed"))
+		}
+		t.Step("bcrypt.GenerateFromPassword")
+
+		hashes = append([]string{string(hash)}, hashes...)
+	}
+
+	needsRevoke := req.Spec.RevokeOldSecrets && len(hashes) > 0
+	if needsRevoke {
+		hashes = []string{hashes[0]}
+	}
+
+	if req.Spec.GenerateNewSecret || needsRevoke {
+		// each bcrypt comparison is expensive and we do not want a large list to cause wasted CPU
+		if len(hashes) > 5 {
```

Comment:
Note that `5` is pretty aggressive here.  You shouldn't really need more than `2`.  (OKTA only allows 2).
At 5 this call could easily take `1 second`.  Continuing to generate these will significantly slow down client authentication.

Reply:
A user accidentally calling this too many times could find themselves in a bad spot, where they can't make new secrets but didn't record new ones, so will have trouble doing a rotation.

The easy solution is to then make an entirely new OIDC client.  This should likely be `documented`.

Reply:
(This feature needs a lot of docs)

Reply:
Note: create a new `page` in the `/site` directory for `docs` for this feature. It probably needs quite a few docs.
The docs will be part of the dynamic clients work (this is merged into a long running branch, not the main branch). Docs will be needed when this feature is complete.